pkg/randchar/generator_test.go

```diff
@@ -0,0 +1,430 @@
+			base:   Alphanumeric,
+			minima: []minimum{{2, Numeric}},
+			n:      3,
+		},
+		"n == min": {
+			base:   Alphanumeric,
+			minima: []minimum{{2, Numeric}},
+			n:      2,
+		},
+		"n < min": {
+			base:                  Alphanumeric,
+			minima:                []minimum{{3, Numeric}},
+			n:                     2,
+			expectedGenerateError: errors.New("n cannot be smaller than the minimum required length of the generator"),
+		},
+		"min not in base set": {
+			base:              Alphanumeric,
+			minima:            []minimum{{2, Symbols}},
+			expectedInitError: errors.New("minimum character set must be a subset of the base character set"),
+			n:                 3,
+		},
+		"empty min set": {
+			base:              Alphanumeric,
+			minima:            []minimum{{2, nil}},
+			expectedInitError: errors.New("minimum character set cannot be empty"),
+			n:                 3,
+		},
+		"zero minimum": {
+			base:              Alphanumeric,
+			minima:            []minimum{{0, Numeric}},
+			expectedInitError: errors.New("minimum must be at least 1"),
+			n:                 3,
+		},
+		"multiple minima": {
+			base:   Alphanumeric,
+			minima: []minimum{{1, Numeric}, {1, Uppercase}},
+			n:      3,
+		},
+		"multiple minima == n": {
+			base:   Alphanumeric,
+			minima: []minimum{{2, Numeric}, {1, Uppercase}},
+			n:      3,
+		},
+		"multiple minima > n": {
+			base:                  Alphanumeric,
+			minima:                []minimum{{2, Numeric}, {2, Uppercase}},
+			n:                     3,
+			expectedGenerateError: errors.New("n cannot be smaller than the minimum required length of the generator"),
+		},
+		"multiple minima with same charset small first": {
+			base:   Alphanumeric,
+			minima: []minimum{{1, Numeric}, {2, Numeric}},
+			n:      3,
+		},
+		"multiple minima with same charset big first": {
+			base:   Alphanumeric,
+			minima: []minimum{{2, Numeric}, {1, Numeric}},
+			n:      3,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			options := make([]Option, len(tc.minima))
+			for i, min := range tc.minima {
+				options[i] = Min(min.count, min.charset)
+			}
+
+			rand, err := NewRand(tc.base, options...)
+			assert.Equal(t, err, tc.expectedInitError)
+
+			// skip other assertions if error case
+			if err != nil {
+				return
+			}
+
+			actual, err := rand.Generate(tc.n)
+			assert.Equal(t, err, tc.expectedGenerateError)
+
+			// skip other assertions if error case
+			if err != nil {
+				return
+			}
+
+			for _, min := range tc.minima {
+				count := countFromSet(actual, min.charset)
+				if count < min.count {
+					t.Errorf("unexpected count for %v: %d (actual) < %d (expected minimum)", min.charset, count, min.count)
+				}
+			}
+		})
+	}
+}
+
+func TestMultipleMin(t *testing.T) {
+	cases := map[string]struct {
+		base                  Charset
+		minima                []minimum
+		expectedInitError     error
+		n                     int
+		expectedGenerateError error
+	}{}
```

Comment:
No cases; Is this still WIP?

Reply:
Was tested elsewhere but forgot to remove this test function.